multicluster/cmd/check.go

```diff
@@ -16,6 +16,7 @@ import (
 	"github.com/linkerd/linkerd2/pkg/servicemirror"
 	"github.com/linkerd/linkerd2/pkg/tls"
 	public "github.com/linkerd/linkerd2/viz/metrics-api/gen/viz"
+	vizHealthCheck "github.com/linkerd/linkerd2/viz/pkg/healthcheck"
```

Comment:
Why does the mutlicluster check command use a viz healthchecker?

Reply:
Because the `Gateways()` API hits prometheus to retrieve the liveness, latency and service count data.

Reply:
Ahh, but I think this means that `multicluster check` will fail if the linkerd-viz extension isn't installed?  Ideally the `multicluster check` would only perform the gateways check if the linkerd-viz extension is installed, and skip that particular check otherwise.

Reply:
Here's the error I get when running `mc check` without first installing viz:

```
 ✗ l mc check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x23654f2]

goroutine 1 [running]:
github.com/linkerd/linkerd2/viz/pkg/healthcheck.(*HealthChecker).vizCategory.func1(0x2aacfe0, 0xc00035ea20, 0x6fc23ac00, 0x2aacfe0)
	/linkerd-build/viz/pkg/healthcheck/healthcheck.go:65 +0x72
github.com/linkerd/linkerd2/pkg/healthcheck.(*HealthChecker).runCheck(0xc00053e840, 0x27b5a0b, 0xb, 0xc00071b9f0, 0xc000890030, 0x100c000)
	/linkerd-build/pkg/healthcheck/healthcheck.go:1506 +0x14c
github.com/linkerd/linkerd2/pkg/healthcheck.(*HealthChecker).RunChecks(0xc00053e840, 0xc000890030, 0xc000890030)
	/linkerd-build/pkg/healthcheck/healthcheck.go:1470 +0x298
github.com/linkerd/linkerd2/viz/pkg/healthcheck.(*HealthChecker).RunChecks(0xc00085ff80, 0xc000890030, 0x4)
	/linkerd-build/viz/pkg/healthcheck/healthcheck.go:54 +0x38
github.com/linkerd/linkerd2/pkg/healthcheck.runChecksTable(0x2a655a0, 0xc00000e018, 0x2a64000, 0xc00085ff80, 0x0)
	/linkerd-build/pkg/healthcheck/healthcheck_output.go:82 +0x14e
github.com/linkerd/linkerd2/pkg/healthcheck.RunChecks(0x2a655a0, 0xc00000e018, 0x2a655a0, 0xc00000e020, 0x2a64000, 0xc00085ff80, 0x27add7c, 0x5, 0x0)
	/linkerd-build/pkg/healthcheck/healthcheck_output.go:36 +0x68
github.com/linkerd/linkerd2/multicluster/cmd.configureAndRunChecks(0x2a655a0, 0xc00000e018, 0x2a655a0, 0xc00000e020, 0xc0004a96a0, 0xc00029eb00, 0x0)
	/linkerd-build/multicluster/cmd/check.go:125 +0x580
github.com/linkerd/linkerd2/multicluster/cmd.newCmdCheck.func1(0xc000927b80, 0x3938b60, 0x0, 0x0, 0x0, 0x0)
	/linkerd-build/multicluster/cmd/check.go:92 +0x219
github.com/spf13/cobra.(*Command).execute(0xc000927b80, 0x3938b60, 0x0, 0x0, 0xc000927b80, 0x3938b60)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x38ec200, 0x0, 0x24a9a20, 0xc00005c0b8)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/linkerd-build/cli/main.go:10 +0x2d
```

Reply:
Done, as describe below :+1: